rxandroidaudio/src/main/java/com/github/piasy/rxandroidaudio/RxAudioPlayer.java

```diff
@@ -86,54 +89,86 @@ public int progress() {
      * play audio from local file. should be scheduled in IO thread.
      */
     public Single<Boolean> play(@NonNull final PlayConfig config) {
-        if (config.mType == PlayConfig.TYPE_FILE && config.mAudioFile != null &&
-                config.mAudioFile.exists()) {
-            return Single.create(new Single.OnSubscribe<Boolean>() {
-                @Override
-                public void call(final SingleSubscriber<? super Boolean> singleSubscriber) {
-                    stopPlay();
+        switch (config.mType) {
```

Comment:
I'm just wondering whether we could encapsulate the parameter check into `PlayConfig` it self, e.g. `isArgumentIllegal()`, so we don't need check it inside the switch, and no more fall through in each case.

If you don't want to do this update, it's totally ok, I'll clean it up and cut a new release :)